src/util.ts

```diff
@@ -76,13 +76,26 @@ export enum ClientSecurity {
 function createClientCreds(
   endpoint: string,
   token: string,
-  security: ClientSecurity = ClientSecurity.SECURE
+  security: ClientSecurity | Buffer = ClientSecurity.SECURE
 ): grpc.ChannelCredentials {
   const metadata = new grpc.Metadata();
   metadata.set("authorization", "Bearer " + token);
 
   const creds = [];
 
+  let rootCert: Buffer | undefined;
+  switch(security)
+  {
+    case ClientSecurity.SECURE:
+    case ClientSecurity.INSECURE_LOCALHOST_ALLOWED:
+    case ClientSecurity.INSECURE_PLAINTEXT_CREDENTIALS:
+      break;
+    default: // security is a Buffer
```

Comment:
I wonder if we can use a type check here somehow... my concern down the road is the use of the overloading behavior

Reply:
I can add a typecheck to see if it's a `Buffer`. Typescript should already catch this mistake though.

The credential generation should throw if it's an invalid certificate inside of `grpc.credentials.createSsl`.

Another alternative is to have add `function createClientCredsWithCert` and make it never overload?

What are you looking?

Reply:
@samkim Thoughts?

Reply:
Yeah I'd prefer the latter; specific functions instead of overloading. We also do something similar in our grpc utils package for our official go client:
https://github.com/authzed/grpcutil/blob/main/dial_opts.go#L48

Reply:
Added a couple of new functions.
- `util.createClientCredsWithCustomCert`
- `NewClientWithCustomCert`
- `NewClientWithChannelCredentials` <-- I exported this one cause it feels like it adds a bunch of flexibility if somebody wants to use it. it would have solved my issues but, we're here, so I left the shorthand version.